pkg/pillar/cmd/zedrouter/acl.go

```diff
@@ -624,7 +640,9 @@ func aceToRules(bridgeName string, vifName string, ace types.ACE, ipVer int, bri
 		outArgs = append(outArgs, []string{"-j", "ACCEPT"}...)
 		inArgs = append(inArgs, []string{"-j", "ACCEPT"}...)
 	}
-	rulesList = append(rulesList, outArgs, inArgs)
+	aclRule3.Rule = inArgs
+	aclRule4.Rule = outArgs
+	rulesList = append(rulesList, aclRule4, aclRule3)
 	if foundLimit {
 		// Add separate DROP without the limit to count the excess
```

Comment:
This limit + unlimited drop rule order assumes -A to add in order. Since you're doing -I in reverse order any limit rule in the manifest will become an unconditional drop.

While one can switch the order of those in this function and continue with -I in reverse order, this indicates that 1) we don't have automated tests for limit rules (not a surprise to me) and 2) you haven't done sufficient manual testing for these changes.

What's the best way to proceed? One way would be to go back to -A and with dropRules doing all the drops at the end (hence no drop rules in the networkinstance code). But that seems like the wrong direction.
Perhaps writing the automated tests for testing accept, limit, portmap, etc rules at this point in time so we can test with these changes in the best path forward?

Reply:
Sure, I am going to pick up test cases for ACLs, in coming sprint.